### PR TITLE
Create Empty from EmptyK

### PR DESCRIPTION
**Pull request by joan38**
This is solving the case where we have an instance of `Empty[T]` but looking for a `Empty[Option[T]]`:
https://scastie.scala-lang.org/Jv24ybPQRQeQ2IEZy5azmA

### CODE REVIEW

**Comment by reardonj** [comment]
Fixes #3617

**Comment by joan38** [comment]
The tests seem flaky, I think we need a maintainer to rerun only the failed ones

File: alleycats-core/src/main/scala/alleycats/Empty.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ object Empty extends EmptyInstances0 {
   def apply[A](a: => A): Empty[A] =
     new Empty[A] { lazy val empty: A = a }
 
-  def fromEmptyK[F[_], T](implicit ekf: EmptyK[F]): Empty[F[T]] = ekf.synthesize[T]
+  implicit def fromEmptyK[F[_], T](implicit ekf: EmptyK[F]): Empty[F[T]] = ekf.synthesize[T]
```
[the remark this answers is not in the document]
Comment (joan38) [reply alleycats-core/src/main/scala/alleycats/Empty.scala]: Pushed with your suggestion.
MIMA should tell us if it's not ok to move the function right?

[the remark this answers is not in the document]
Comment (joan38) [reply alleycats-core/src/main/scala/alleycats/Empty.scala]: @armanbilge could you rerun the failing instances please? :)

**Comment by rossabaker** [comment]
Error was:

```
  [error] Test runner interrupted by fatal signal 11
  [warn] Force close java.lang.RuntimeException: Process /home/runner/work/cats/cats/tests/native/target/scala-2.13/cats-tests-test finished with non-zero value 139 (0x8b)
```

Rerunning.

**Comment by joan38** [comment]
@rossabaker @armanbilge Good to go?

**Comment by rossabaker** [comment]
Code looks good.  Would it be possible to add a quick test, similar to what's in that Scastie, just to demonstrate that it compiles now?  Or even just the `Empty[Option[LocalDateTime]]` example from the original ticket?

**Comment by joan38** [comment]
Added a test but open to suggestions on improving it

File: alleycats-core/src/test/scala/alleycats/SyntaxSuite.scala [hunk 2]
```diff
@@ -54,6 +54,21 @@ object SyntaxSuite {
...
 
+  def testOptionEmpty[A: Empty]: Unit = {
+    case class A[T](x: T)
+    case class B()
+
+    import alleycats.std.option.*
+    implicit def emptyA[T: Empty]: Empty[A[T]] = new Empty[A[T]] {
+      def empty: A[T] = A(Empty[T].empty)
+    }
+    implicit val emptyB: Empty[B] = new Empty[B] {
```
Comment (johnynek) [inline alleycats-core/src/test/scala/alleycats/SyntaxSuite.scala:65]: You don't need this do you? Shouldn't we have an empty `Option[B]` for all `B`?

File: alleycats-core/src/test/scala/alleycats/SyntaxSuite.scala [hunk 3]
```diff
@@ -54,6 +54,21 @@ object SyntaxSuite {
     val a1: Boolean = x.nonEmpty
   }
 
+  def testOptionEmpty[A: Empty]: Unit = {
```
Comment (johnynek) [inline alleycats-core/src/test/scala/alleycats/SyntaxSuite.scala:57]: Where is this type parameter used? I don't see it. Can we remove it?

Comment (joan38) [reply alleycats-core/src/test/scala/alleycats/SyntaxSuite.scala:57]: Not sure why I did that 🤦🏻‍♂️

File: alleycats-core/src/test/scala/alleycats/SyntaxSuite.scala (hunk 2 above)
Comment (joan38) [reply alleycats-core/src/test/scala/alleycats/SyntaxSuite.scala]: That right. I pushed

**Review by johnynek — APPROVED** [review]

**Comment by joan38** [comment]
@rossabaker Feel free to merge it. Thanks everyone

**Comment by joan38** [comment]
Should we merge this?

**Review by rossabaker — APPROVED** [review]
